x-pack/platform/plugins/private/data_visualizer/public/lite/file_upload_lite_action.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { Trigger, UiActionsActionDefinition } from '@kbn/ui-actions-plugin/public';
+import type { CoreSetup } from '@kbn/core/public';
+import {
+  OPEN_FILE_UPLOAD_LITE_ACTION,
+  OPEN_FILE_UPLOAD_LITE_TRIGGER,
+} from '@kbn/file-upload-common';
+
+import type { OpenFileUploadLiteContext } from '@kbn/file-upload-common';
+import type { DataVisualizerStartDependencies } from '../application/common/types/data_visualizer_plugin';
+
+export const createOpenFileUploadLiteTrigger: Trigger = {
+  id: OPEN_FILE_UPLOAD_LITE_TRIGGER,
+  title: i18n.translate('xpack.dataVisualizer.file.lite.actions.triggerTitle', {
+    defaultMessage: 'Open file upload UI',
+  }),
+  description: i18n.translate('xpack.dataVisualizer.file.lite.actions.triggerDescription', {
+    defaultMessage: 'Open file upload UI',
+  }),
+};
+
+export function createOpenFileUploadLiteAction(
+  getStartServices: CoreSetup<DataVisualizerStartDependencies>['getStartServices']
+): UiActionsActionDefinition<OpenFileUploadLiteContext> {
+  return {
+    id: 'create-open-file-upload-lite-action',
+    type: OPEN_FILE_UPLOAD_LITE_ACTION,
+    getIconType(context): string {
+      return 'machineLearningApp';
+    },
+    getDisplayName: () =>
+      i18n.translate('xpack.dataVisualizer.file.lite.actions.displayName', {
+        defaultMessage: 'Open file upload UI',
+      }),
+    async execute({
+      onUploadComplete,
+      autoAddInference,
+      indexSettings,
+    }: OpenFileUploadLiteContext) {
+      try {
+        const [{ showFlyout }, [coreStart, { share, data }]] = await Promise.all([
+          import('./flyout/show_flyout'),
+          getStartServices(),
+        ]);
+
+        await showFlyout(coreStart, share, data, {
```

Comment:
Flyout should be opened without `await` so there is immeditate user feedback for the action. The flyout contents should be async imported.

Something like the below

```
execute: async (context) => {
    const LazyModal = lazy(async () => {
      const { ModalContent } = await import('./modal_content');
      return {
        default: ModalContent,
      };
    });
    coreServices.overlays.openModal(
      toMountPoint(
        <Suspense fallback={<EuiSkeletonText />}>
          <LazyModal
            context={context}
          />
        </Suspense>,
        coreServices
      )
    );
  }
```

Reply:
Good point. Updated in https://github.com/elastic/kibana/pull/206864/commits/f0cad9bdc7cabfad6451c375c06c03ea850d3e96

There is some code clean up which is needed inside the flyout contents which can be done in a follow up PR. but for now the flyout is opened as soon as the trigger is invoked and the contents loaded async.